pkg/storaged/crypto-keyslots.jsx

```diff
@@ -355,67 +342,67 @@ function edit_tang_adv(client, block, key, url, adv, passphrase) {
-                        </div>
-                    </div>
-                </div>
-            );
-        }
-    };
-};
+        this.onRemove = this.onRemove.bind(this);
+    }
 
-function remove_passphrase_dialog(block, key) {
-    dialog_open({
-        Title: _("Remove passphrase"),
-        Fields: [
-            RemovePassphraseField("passphrase", key, block_name(block))
-        ],
-        Action: {
-            DangerButton: true,
-            Title: _("Remove"),
-            action: function (vals) {
-                if (vals.passphrase === false)
-                    return slot_remove(block, key.slot);
-                else
-                    return passphrase_remove(block, vals.passphrase);
-            }
-        }
-    });
+    onRemove() {
+        this.setState({ pending: true });
+        const dev = decode_filename(this.props.block.Device);
+        const cmd = ["cryptsetup", "luksKillSlot", dev, this.props.slot.toString()];
+        if (this.state.force)
+            cmd.splice(2, 0, "-q");
+
+        const spawn = cockpit.spawn(cmd, { superuser: true, err: "message" });
+        if (!this.state.force)
+            spawn.input(this.state.passphrase + "\n");
+
+        spawn.then(this.props.onClose)
+                .catch(e => this.setState({ error: e.message }))
+                .always(() => this.setState({ pending: false }));
+    }
+
+    render() {
+        return (
+            <Modal
+                title={cockpit.format(_("Remove passphrase in key slot $0"), this.props.slot)}
+                titleIconVariant="warning"
+                isOpen
+                onClose={this.props.onClose}
+                footer={<>
+                    {
+                        this.state.error && <ModalError dialogError={this.state.error} />
+                    }
+                    <Button key="remove" variant="danger" isLoading={this.state.pending} isDisabled={this.state.pending} onClick={this.onRemove}>{_("Remove")}</Button>
+                    <Button key="cancel" variant="link" isDisabled={this.state.pending} onClick={this.props.onClose}>{_("Cancel")}</Button>
+                </>}
+                position="top" variant="medium"
+            >
+                <Stack hasGutter>
+                    <p>{ fmt_to_fragments(_("Passphrase removal may prevent unlocking $0."), <b>{block_name(this.props.block)}</b>) }</p>
+                    <Checkbox id="run-image-dialog-tty"
+                              isChecked={!this.state.force}
+                              label={_("Confirm removal")}
+                              onChange={checked => this.setState({ force: !checked })}
+                              body={this.state.force
+                                  ? <p className="slot-warning">
+                                      {_("Removing a passphrase without confirmation can be dangerous.")}
+                                  </p>
+                                  : <Form isHorizontal>
```

Comment:
Conceptually the Form should contain also the Checbox. It's part of the form right :) ?

Reply:
Not really - this form is part of checkbox body

Reply:
Right, but what I saw is that it's conceptually wrong to leave some fields outside of the form. In this case the checkbox. 

Copying from https://www.patternfly.org/v4/components/form/design-guidelines: 
A form is a group of elements used to collect information from a user in a variety of contexts including in a modal

Whenever we use forms in modals we put all fields of the modals inside the form. I don't understand why this is special.